front/hooks/useTableOrDocFetcher.ts

```diff
@@ -0,0 +1,90 @@
+import { useCallback, useEffect, useState } from "react";
+
+interface TableOrDoc {
+  name: string;
+  description?: string;
+  text?: string;
+  tags?: string[];
+  sourceUrl?: string;
+}
+
+interface Owner {
+  sId: string;
+}
+
+interface DataSourceView {
+  dataSource: {
+    name: string;
+  };
+}
+
+interface Table {
+  name: string;
+  description: string;
+}
+
+export const useTableOrDocFetcher = (
+  initialId: string | null,
+  isTable: boolean,
+  table: Table | null,
+  owner: Owner,
+  dataSourceView: DataSourceView,
+  resetTableOrDoc: () => void
+) => {
+  const [tableOrDoc, setTableOrDoc] = useState<TableOrDoc>({
+    name: "",
+  });
+  const [isLoading, setIsLoading] = useState(false);
+  const [uploading, setUploading] = useState(false);
+
+  const fetchData = useCallback(async () => {
+    if (!initialId) {
+      resetTableOrDoc();
+      return;
+    }
+    setIsLoading(true);
+    if (isTable && table) {
+      setTableOrDoc((prev) => ({
+        ...prev,
+        name: table.name,
+        description: table.description,
+      }));
+    } else {
+      try {
+        const res = await fetch(
+          `/api/w/${owner.sId}/data_sources/${
+            dataSourceView.dataSource.name
+          }/documents/${encodeURIComponent(initialId)}`
+        );
+        if (res.ok) {
+          const document = await res.json();
+          setTableOrDoc((prev) => ({
+            ...prev,
+            name: initialId,
+            text: document.document.text,
+            tags: document.document.tags,
+            sourceUrl: document.document.source_url,
+          }));
+        }
+      } catch (e) {
+        console.error(e);
+      } finally {
+        setUploading(false);
```

Comment:
this uploading status is definitely not clear, we are not uploading anything, setting to false something that is already false, and return an uploading status we don't use :-)